synapseclient/client.py

```diff
@@ -1628,6 +1629,26 @@ def store(
                     ] or not self.cache.contains(
                         bundle["entity"]["dataFileHandleId"], entity["path"]
                     )
+
+                    # Check if we got an MD5 checksum from Synapse and compare it to the local file
+                    if (
+                        (entity["synapseStore"] or local_state.get("synapseStore"))
+                        and os.path.isfile(entity["path"])
+                        and needs_upload
+                        and (
+                            md5_stored_in_synapse := (
+                                local_state.get("_file_handle", {}).get(
+                                    "contentMd5", None
+                                )
+                                or (fileHandle or {}).get("contentMd5", None)
+                            )
+                        )
+                    ):
```

Comment:
The logic in this function is pretty complicated/difficult to read. Would it be possible to break up some of these logical statements or pull some out of this `if` block?

Reply:
I modified this if check a bit to make it a bit easier to read - I also verified that `md5_for_file` utility function is still only getting called once and `local_file_md5_hex` is getting pass along to the `upload_file_handle` function so it doesn't get re-calculated again later on